### PR TITLE
fix reward clipping in dynamic_action_repeat_agent

### PR DESCRIPTION
**Pull request by hnyu**
Bug introduced in #707 . The current clipping is performed on any accumulated reward. Because the number of accumulation steps can vary, this clipping value is not ideal. To have similar semantics with reward clipping (on individual rewards) in other typical algorithms, now the clip value is multiplied with `repeats`.  Although this fix is still not equivalent to first clipping individual rewards and then accumulating them, it's done for code simplicity. This fix improves the algorithm's performance.

### CODE REVIEW

File: alf/algorithms/dynamic_action_repeat_agent.py [hunk 1]
```diff
@@ -346,7 +345,10 @@ def preprocess_experience(self, rl_exp):
 
             clip = self._reward_normalizer._clip_value
             if clip > 0:
-                reward = torch.clamp(reward, -clip, clip)
+                # The clip value is for single-step rewards, so we need to multiply
+                # it with the repeated steps.
+                clip = clip * repeats
+                reward = torch.max(torch.min(clip, reward), -clip)
```
Comment (emailweixu) [inline alf/algorithms/dynamic_action_repeat_agent.py:354]: why not keep using clamp()?

Comment (hnyu) [reply alf/algorithms/dynamic_action_repeat_agent.py:354]: > why not keep using clamp()?

clamp only supports scalar min and max as I know. After multiplying `repeats`, `clip` is a tensor.

**Review by emailweixu — APPROVED** [review]
